### PR TITLE
Fix timer

### PR DESCRIPTION
**Pull request by eginhard**
## Description
These changes fix various warnings and errors that occurred when using the timer and prevented it from ringing.

## How Has This Been Tested?
I confirmed that the warnings disappeared and the timer can now ring, using both gstreamer or pyglet if that is available.

There is a threading issue when using pyglet and closing Gourmet after the timer has gone off, but I don't know how to solve that for now.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: gourmet/sound_gst.py [hunk 1]
```diff
@@ -7,13 +7,14 @@
...
 
 class Player:
-    def __init__ (self):
-        self.player = Gst.ElementFactory.make('playbin2','player')
+    def __init__(self):
+        Gst.init()
+        self.player = Gst.ElementFactory.make('playbin', 'player')
 
-    def play_file (self,path):
+    def play_file(self,path):
```
Comment (cydanil) [inline gourmet/sound_gst.py]:
```suggestion
    def play_file(self, path: str):
```

File: gourmet/sound_gst.py [hunk 2]
```diff
@@ -7,13 +7,14 @@
...
         self.player.set_state(Gst.State.NULL)
-        self.player.set_property('uri','file://'+path)
+        self.player.set_property('uri', 'file://'+path)
         self.player.set_state(Gst.State.PLAYING)
 
-    def stop_play (self,path):
+    def stop_play(self,path):
```
Comment (cydanil) [inline gourmet/sound_gst.py]:
```suggestion
    def stop_play(self):
```

File: requirements.txt [hunk 3]
```diff
@@ -6,6 +6,7 @@ lxml
 Pillow
 pycairo
 pyenchant
+pyglet
```
Comment (cydanil) [inline requirements.txt]: Nice catch :) 
Could you also add a `python3-pyglet` entry in the [INSTALL.md](https://github.com/kirienko/gourmet/blob/master/INSTALL.md#unbutu-2004-linux-mint-20) ?

**Review by cydanil — CHANGES_REQUESTED** [review]
Thanks for taking the time to work on this!
I think that you can mitigate the threading issue with the following:

```python
Class Player:
    def __init__(self):
        Gst.init()
        self.player = Gst.ElementFactory.make('playbin', 'player')
        self.player.connect('about-to-finish', self.stop_play)

     def stop_play(self):
           self.player.set_state(Gst.State.NULL)
```

Would you mind testing that?

Also, could you clean the imports such that it's
```python
from gi.repository import Gst
```
and add to `gourmet/__init__.py`
```python
require_version("Gst", "1.0")
```

File: gourmet/sound_gst.py (hunk 2 above)
Comment (eginhard) [reply gourmet/sound_gst.py]: I removed the `stop_play` methods as they are not used.

**Comment by eginhard** [comment]
Thanks, I made those changes. With `sound_gst.py` (when pyglet is not available) there are no issues, it's from `sound_pyglet.py`. After pyglet has played the timer (the sound also seems cut off compared to gst) and I want to close Gourmet, a dialog with "Another process is in progress - Exit program anyway?" appears and selecting "Quit" logs the following:
```
DEBUG: gourmet/gourmet/GourmetRecipeManager.py: 466 Unable to terminate thread <GLibMainLoopThread(Thread-1, started daemon 140028434503424)>
```

Are all the different ways to play sound actually necessary (there's also `sound_windows.py`) or is there something that works well across platforms?

File: gourmet/__init__.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 from gi import require_version
 require_version("Gdk", "3.0")
+require_version('Gst', '1.0')
```
Comment (cydanil) [inline gourmet/__init__.py]: A bit of a nitpick here... Could you use either double quotes or single quotes for all strings in that file?

**Comment by cydanil** [comment]
I think only Gst should be used: the historical reason for having different players was that back in the days, sound handling was not that well integrated between user applications and the OS.

Indeed, you could remove the other sound handlers.

**Comment by eginhard** [comment]
Great, that simplified things.

**Review by cydanil — APPROVED** [review]
Cracking, thanks a lot!

**Pull request merged**